### ISSUE TITLE
Remove Service monkey patch

### ISSUE THREAD

**Issue by yoshiwaan** [rackspace-cookbooks/varnish#85]
Following up from https://github.com/rackspace-cookbooks/varnish/pull/83
I was asking about safer solutions for the desired functionality on the varnish logging services:
 * Sleep 5 before attempting
 * Retry 5 times
And was made aware of the following common functionality of Chef resources:
https://docs.chef.io/resource_common.html#properties
Specifically the following:
* retries
* retry_delay
These can be added to any Chef resource to make multiple attempts on failure and add a delay between restarts if retrying. Thus the logging library (libraries/varnish_log.rb) can update the service declaration at the bottom of the configure_varnish_log method to allow the same functionality without the need of a monkey patch
```
service new_resource.log_format do
  supports restart: true, reload: true
  action %w(enable start)
  retries 5
  retry_delay 5
end
```
if an initial delay is required before the first run a ruby block can run a sleep
```
ruby_block do
  block do
    sleep(5)
  end
  action :run
end
```

**Comment by yoshiwaan** [comment]
I can make a PR if desired

**Comment by martinb3** [comment]
@yoshiwaan -- that would be great, if you don't mind.

**Comment by yoshiwaan** [comment]
@martinb3 https://github.com/rackspace-cookbooks/varnish/pull/86
I've run a test on a AWS instance via kitchen and it seemed fine, although I've not run it through the full kitchen list

**Comment by martinb3** [comment]
@yoshiwaan please see my comments on #86. The workaround seems to be failing on some of the test suites.

**Comment by martinb3** [comment]
Fixed in #86.

**Comment by yoshiwaan** [comment]
Thanks very much @martinb3
